skll/data/readers.py

```diff
@@ -715,13 +717,34 @@ def _sub_read(self, f):
         features : list of dicts
             The features for the features set.
         """
-        try:
-            df = pd.read_csv(f, engine='c')
-        except CParserError:
-            df = pd.read_csv(f)
+        df = pd.read_csv(path, sep=self._sep, engine='c')
```

Comment:
So the 'c' engine would never fail for this?

Reply:
I couldn't come up with a scenario where (1) the 'c' engine would fail, and (2) the 'python' engine wouldn't fail, _unless_ it's a situation where the user needs to pass additional arguments to `read_csv()` that aren't available with the 'c' parser.

If we want to allow users to pass additional arguments directly to the `pandas` reader, then I think this makes sense, but since we're not allowing them to do that, I don't see the advantage.

Let me know if you think I'm missing something.

Reply:
Perhaps that would be a nice thing to have to allow for keyword argument to `pandas` in case we just can't think of a solution. Would it be too much work to add that?

Reply:
Well, it depends. :)

We could  just pass any additional keyword arguments to the `Reader` constructor onto `pd.read_csv()` (or whatever the `pandas` I/O function is). Alternatively, we could allow users to pass arguments to the constructor with a prefix, like `pandas_*` (e.g. `pandas_engine='python'`.), or something like that.

My only concern is that there are a lot of possible `pandas` arguments, so as long as we can be agnostic about what the `pandas` arguments are (and not have to test each one), this is easy.

Does that make sense? Thoughts?

Reply:
I think we can follow the example of the `Learner` class where we have `model_kwargs` that are keyword arguments are just passed on to the underlying initializer. That should be simple enough right? So something like `pandas_kwargs`?

Reply:
Yup! Sounds good. I'll update.

Reply:
Thanks for your continued patience :)